src/services/forge/forge.service.ts

```diff
@@ -0,0 +1,48 @@
+import _ from "lodash";
+import Forges from "../../data/models/forge-schema";
+import { IForge } from "../../types/forge/forge";
+import { ICharacter } from "../../types/character/character";
+import { Document } from "mongoose";
+
+export class ForgeService {
+  private static _instance: ForgeService;
+
+  public static getInstance(): ForgeService {
+    if (_.isNil(ForgeService._instance)) {
+      ForgeService._instance = new ForgeService();
+    }
+    return ForgeService._instance;
+  }
+
+  public getEntity(bossId: ICharacter["ownerId"]): Promise<IForge | void> {
```

Comment:
I never saw this syntax before but I do like it.
I do not know if there is a drawback from this though.
In fact, this could give a LOT of reworking.

Reply:
Oh... I see. In a future PR? 😄 I think I see what reworking I would need to do... 😛 

It's because I wanted to type it, but to say really: "You will receive a particular string." which is nicer than plain string xD

Reply:
@asmodee-bellanie I did not ask you to change it.
On the contrary I think that this is a good syntaxe I did not knew about but in the other hand since I did not saw this before I am just wondering if this is good.
Because you can do it does not mean that this is right but I still really think that this is good but not everywhere.

Reply:
I wasn't saying I would change this one... but all the "memberId" I use to a syntax like that. In an other PR.

I've seen this there: https://medium.com/@tomanagle/strongly-typed-models-with-mongoose-and-typescript-7bc2f7197722

It's to type more strongly your data.